internal/service/manager_test.go

```diff
@@ -0,0 +1,108 @@
+package service_test
+
+import (
+	"testing"
+
+	"github.com/windmilleng/tilt/internal/model"
+	"github.com/windmilleng/tilt/internal/service"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAdd(t *testing.T) {
+	f := newServiceManagerFixture(t)
+
+	s := model.Service{Name: model.ServiceName("hello")}
+	err := f.sm.Add(s)
+	assert.NoError(t, err)
```

Comment:
i don't think this stops function execution if there IS an error, which will result in confusing nil pointer shit further along in execution. should do:
```
if assert.NoError(t, err) {
    <rest of test>
}
```
(also, noting the func that threw the error (`assert.NoError(t, err, "ListServices")` is A+ helpful in debugging if you feel like doing it.)

Reply:
Good call!